staker-rewards/README.adoc

```diff
@@ -1,35 +1,112 @@
 = Staker rewards
 
-== Prerequisites
+ECDSA staker rewards are based on ETH locked and KEEP staked over time. The
+rewards calculations are performed weekly off-chain, and committed for
+distribution using a variant of Uniswap’s Merkle distributor. This repository
+contains scripts for calculating rewards and generation of Merkle tree which
+can be used by stakers to claim their rewards for a given interval. Anyone
+can use these scripts to validate reward distribution or to generate reward
+estimates for a pending interval.
 
-To run the script, make sure you have the following prerequisites:
+== Algorithm
 
-- Node.js (at least v14.3.0). We recommend to use
-  the https://github.com/nvm-sh/nvm[Node Version Manager] to get the right
-  Node.js version with no troubles. You can install this tool by running
-  `brew install nvm` on macOS or follow the https://github.com/nvm-sh/nvm#installing-and-updating[installation guide]
-  for other operating systems.
+The algorithm for calculating ECDSA staker rewards as implemented in the scripts 
+in this repository is described in this section.
 
-== How to run
+=== Input
+* Interval `T` starts at `T_start`.
 
-1. Make sure you use the right Node.js version:
-+
-```
-nvm use 14.3.0
-```
-2. Install the NPM dependencies if not done yet:
-+
+* Interval `T` ends at `T_end`.
+
+* All operators who have ever bonded ETH are taken into account.
+
+=== Operator Parameters
+For each Operator:
+
+* Calculate the amount of KEEP staked at `T_end` as `KEEP_staked`.
+
+* Calculate the total amount of ETH bonded at `T_start` as `ETH_bonded`.
+
+* Read the amount of ETH unbonded at `T_start` as `ETH_unbonded`.
+
+* Calculate the sum of ETH unbonded and withdrawn for the entire interval
+  as `ETH_withdrawn`.
+  
+* Calculate the total ETH under management as 
+  `ETH_locked = ETH_bonded + ETH_unbonded - ETH_withdrawn`.
+
+=== Requirements
+
+Operator not meeting all requirements listed in this section is not getting any
+rewards for interval `T`. Rewards proportional to the operator's reward weight
+are deallocated and remains in the pool for future intervals. 
+
+* If the operator does not have `BondedECDSAKeepFactory` contract or tBTC sortition
+  pool authorized at `T_start`, they are not getting any rewards for interval `T`.
+  
+* If the operator does not have a minimum KEEP stake at `T_start`, they are not
+  getting any rewards for interval `T`.
+  
+* If the operator has more `ETH_unbonded` than the minimum unbonded value and is
+  not in the sortition pool at `T_start`, they are not getting any rewards for
+  interval `T`. In other words, if the operator is eligible to be in the
+  sortition pool, they have to be in the sortition pool. Operators who are not
+  in the sortition pool because all of their ETH is bonded are still getting
+  rewards because that ETH is still under the system’s management.
```

Comment:
```suggestion
  interval `T`. In other words, if the operator is eligible to be in the
  sortition pool, they have to be registered in the sortition pool. Operators who are not
  registered in the sortition pool because all of their ETH is bonded are still getting
  rewards because that ETH is still under the system’s management.
```